pkg/replication/admission.go

```diff
@@ -489,16 +497,15 @@ func addAnnotation(key, value string) util.Patch {
 	}
 }
 
-func generateUpdated(operation admissionv1beta1.Operation, newSD *comp_v1.ServiceDescriptor) (time.Time, error) {
-	if operation != admissionv1beta1.Update {
-		return time.Now(), nil
-	}
-	ts, exists := newSD.Annotations[updatedKey]
-	if !exists || ts == "" {
-		return time.Now(), nil
+func generateUpdated(isReplication bool, operation admissionv1beta1.Operation, newSD *comp_v1.ServiceDescriptor) (time.Time, error) {
+	if isReplication {
+		ts, exists := newSD.Annotations[updatedKey]
+		if !exists {
+			return time.Now(), errors.New("replication controller must set the mutation timestamp")
+		}
+		return parseTimestamp(ts)
 	}
-
-	return parseTimestamp(ts)
+	return time.Now(), nil
```

Comment:
Sorry, I think we may need to keep the timestamp if hash hasn't changed, e.g. when someone adds/deletes annotation or finalizer.
So you need to pass a flag if hash has changed?